ofrak_core/test_ofrak/unit/test_ofrak_server.py

```diff
@@ -379,3 +394,267 @@ async def test_add_tag(ofrak_client: TestClient, hello_world_elf):
         json="ofrak.core.apk.Apk",
     )
     assert resp.status == 200
+
+
+async def test_update_script(ofrak_client: TestClient, hello_world_elf):
+    create_resp = await ofrak_client.post(
+        "/create_root_resource", params={"name": "hello_world_elf"}, data=hello_world_elf
+    )
+    root = await create_resp.json()
+    root_id = root["id"]
+
+    # Perform some actions that should hit add_action, add_variable, and all their subcalls
```

Comment:
So there are no unit tests that actually explicitly call `add_action`, `add_variable`, etc.?

Reply:
They don't have to be, and testing GUI interaction with them is necessary too.